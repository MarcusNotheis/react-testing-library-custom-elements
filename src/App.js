import React from "react";
import "./styles.css";
import "@ui5/webcomponents/dist/Button.js";

// class MyWebComponent extends HTMLElement {
//   connectedCallback() {
//     this.innerHTML = `<b>${this.getAttribute("text")}!</b>`;
//   }
// }

// if (!customElements.get("my-webcomponent")) {
//   customElements.define("my-webcomponent", MyWebComponent);
// }

export default function App() {
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <h2>Start editing to see some magic happen!</h2>
      <my-webcomponent text="This is my Web Component" />
      {/* <ui5-button>My Button Text</ui5-button> */}
    </div>
  );
}
