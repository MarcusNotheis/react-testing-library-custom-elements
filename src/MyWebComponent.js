export class MyWebComponent extends HTMLElement {
    anyOtherInstanceAttribute = '123'
    connectedCallback() {
      this.innerHTML = `<b>${this.getAttribute("text")}!</b>`;
    }
  }
  