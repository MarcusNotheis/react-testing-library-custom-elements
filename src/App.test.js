import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
import { MyWebComponent } from "./MyWebComponent";

test("renders learn react link", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test("with web component", () => {
  if (!customElements.get("my-webcomponent")) {
    customElements.define("my-webcomponent", MyWebComponent);
  }
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
