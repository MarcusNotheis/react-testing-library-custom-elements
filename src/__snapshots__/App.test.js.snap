// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Hello CodeSandbox
    </h1>
    <h2>
      Start editing to see some magic happen!
    </h2>
    <my-webcomponent
      text="This is my Web Component"
    />
  </div>
</DocumentFragment>
`;

exports[`with web component 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Hello CodeSandbox
    </h1>
    <h2>
      Start editing to see some magic happen!
    </h2>
    MyWebComponent {
      "anyOtherInstanceAttribute": "123",
    }
  </div>
</DocumentFragment>
`;
